frontend/js/behaviors/layeredImageViewer/layeredImageViewer.js

```diff
@@ -1339,147 +1339,203 @@ const layeredImageViewer = function(container) {
       const maxWidth = container.dataset.maxWidth || parentWidth;
       const width = Math.min(parentWidth, maxWidth);
 
-      const figure = container.querySelector('figure');
       let height = 1;
       let aspect = 0;
 
-      // Calculate dimensions based on figure if available
-      if (figure) {
-        // Get aspect ratio from figure dimensions or data attribute
-        const figureWidth = figure.offsetWidth || figure.naturalWidth || width;
-        const figureHeight = figure.offsetHeight || figure.naturalHeight;
+      const mainFigure = container.querySelector('figure.o-layered-image-viewer__osd');
 
-        if (figureHeight && figureWidth) {
-          aspect = figureWidth / figureHeight;
-          height = width / aspect;
-        } else {
-          // Fallback to data attribute or previous dimensions
-          aspect = parseFloat(container.dataset.aspect ||
-              (containerDimensions ? containerDimensions.aspect : 1));
-          height = width / aspect;
-        }
+      if (mainFigure) {
+        height = calculateHeightFromFigure(mainFigure, width);
+        aspect = width / height;
       } else {
-        // Fallback when no figure element
-        aspect = parseFloat(container.dataset.aspect ||
-            (containerDimensions ? containerDimensions.aspect : 1));
-        height = width / aspect;
+        // LayeredImageViewer not initialized yet - use reasonable default
+        const defaultAspect = 3/4; // Default aspect ratio
+        height = Math.min(width * defaultAspect, 600); // Max 600px height
+        aspect = width / height;
       }
 
-      // Store updated dimensions
-      containerDimensions = {
-          width,
-          height,
-          aspect
-      };
-
-      container.style.width = `${containerDimensions.width}px`;
-      container.style.height = `${containerDimensions.height}px`;
+      // Store and apply dimensions
+      containerDimensions = { width, height, aspect };
+      container.style.width = `${width}px`;
+      container.style.height = `${height}px`;
       container.style.backgroundColor = '#f5f5f5';
+    }
+  };
 
-      // Hide the figure element initially
-      if (figure) {
-          figure.style.display = '';
+  const calculateHeightFromFigure = (figure, containerWidth) => {
+    const mediaImg = figure.querySelector('.m-media__img');
+    const figcaption = figure.querySelector('figcaption');
+
+    if (!mediaImg) {
+      return containerWidth * (3/4); // Fallback
+    }
+
+    const idealImageHeight = containerWidth * (5/4);
+    const computedStyle = window.getComputedStyle(mediaImg);
+    const cssMaxHeight = computedStyle.maxHeight;
+
+    let maxHeight = 650; // Default fallback
```

Comment:
[Line 1353](https://github.com/art-institute-of-chicago/artic.edu/pull/872/files#diff-14aadd230bd3c15af4dcb1f7c94ea621e8ee27171e96cc0a85dd5e1ffb1011a9R1353) says the max height is 600.